src/blockindices.jl

```diff
@@ -203,6 +203,8 @@ BlockIndexRange(block::Block{N}, inds::Vararg{AbstractUnitRange{<:Integer},N}) w
 
 block(R::BlockIndexRange) = R.block
 
+copy(R::BlockIndexRange) = BlockIndexRange(R.block, copy.(R.indices))
```

Comment:
Perhaps we may use `map` instead of broadcasting, as it is a simpler operation for `Tuple`s?

Reply:
In fact I had to change a bunch of broadcast operations to map to avoid mystery unnecessary allocations.

Reply:
Yes I had noticed that and then promptly forgot when I made this PR...